aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -222,6 +228,151 @@ Tensor& addr_out(Tensor &result, const Tensor& self, const Tensor& vec1, const T
   return at::_addr_out(result, self, vec1, vec2, beta, alpha);
 }
 
+template <typename scalar_t>
+inline void bmm_cpu_kernel(const Tensor& result, const Tensor& self, const Tensor& mat2) {
+  int64_t bs = result.size(0);
+  int64_t is = result.size(1);
+  int64_t js = result.size(2);
+  int64_t ks = self.size(2);
+
+  auto r0 = result.accessor<scalar_t, 3>();
+  auto s0 = self.accessor<scalar_t, 3>();
+  auto m0 = mat2.accessor<scalar_t, 3>();
+
+#ifdef _OPENMP
```

Comment:
It might be good to remark that better results may be had by calling MKL, to let a potential future optimizer who is not you try it out.